extension/parquet/geo_parquet.cpp

```diff
@@ -0,0 +1,395 @@
+			yyjson_mut_obj_add(column_json, crs_key, crs_val);
+			yyjson_doc_free(crs_doc);
+		}
+	}
+
+	yyjson_write_err err;
+	size_t len;
+	char *json = yyjson_mut_write_opts(doc, 0, nullptr, &len, &err);
+	if (!json) {
+		yyjson_mut_doc_free(doc);
+		throw SerializationException("Failed to write JSON string: %s", err.msg);
+	}
+
+	// Create a string from the JSON
+	duckdb_parquet::format::KeyValue kv;
+	kv.__set_key("geo");
+	kv.__set_value(string(json, len));
+
+	// Free the JSON and the document
+	free(json);
+	yyjson_mut_doc_free(doc);
+
+	file_meta_data.key_value_metadata.push_back(kv);
+	file_meta_data.__isset.key_value_metadata = true;
+}
+
+bool GeoParquetFileMetadata::IsGeometryColumn(const string &column_name) const {
+	return geometry_columns.find(column_name) != geometry_columns.end();
+}
+
+unique_ptr<ColumnReader> GeoParquetFileMetadata::CreateColumnReader(ParquetReader &reader,
+                                                                    const LogicalType &logical_type,
+                                                                    const SchemaElement &s_ele, idx_t schema_idx_p,
+                                                                    idx_t max_define_p, idx_t max_repeat_p,
+                                                                    ClientContext &context) {
+
+	D_ASSERT(IsGeometryColumn(s_ele.name));
+
+	const auto &column = geometry_columns[s_ele.name];
+
+	// Get the catalog
+	auto &catalog = Catalog::GetSystemCatalog(context);
+
+	// WKB encoding
+	if (logical_type.id() == LogicalTypeId::BLOB && column.geometry_encoding == GeoParquetColumnEncoding::WKB) {
+		// Look for a conversion function in the catalog
+		auto &conversion_func_set =
+		    catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, DEFAULT_SCHEMA, "st_geomfromwkb")
+		        .Cast<ScalarFunctionCatalogEntry>();
+		auto conversion_func = conversion_func_set.functions.GetFunctionByArguments(context, {LogicalType::BLOB});
+
+		// Create a bound function call expression
+		auto args = vector<unique_ptr<Expression>>();
+		args.push_back(std::move(make_uniq<BoundReferenceExpression>(LogicalType::BLOB, 0)));
+		auto expr =
+		    make_uniq<BoundFunctionExpression>(conversion_func.return_type, conversion_func, std::move(args), nullptr);
+
+		// Create a child reader
+		auto child_reader =
+		    ColumnReader::CreateReader(reader, logical_type, s_ele, schema_idx_p, max_define_p, max_repeat_p);
+
+		// Create an expression reader that applies the conversion function to the child reader
+		return make_uniq<ExpressionColumnReader>(context, std::move(child_reader), std::move(expr));
+	}
+
+	// Otherwise, unrecognized encoding
+	throw NotImplementedException("Unsupported geometry encoding");
+}
+
+bool GeoParquetFileMetadata::IsSpatialExtensionInstalled(ClientContext &context) {
+	auto &catalog = Catalog::GetSystemCatalog(context);
+
+	// Look for ST_GeomFromWKB in the catalog
+	// TODO: Check for more functions
```

Comment:
Is this a TODO that was meant to be addressed?

Reply:
Yeah in general just looking for this one function is probably sufficient it was more like a reminder to add more checks once I add more functionality and the complete set of functions required for handling all the geoparquet details are known.

Reply:
Yeah in general just looking for this one function is probably sufficient it was more like a reminder to add more checks once I add more functionality and the complete set of functions required for handling all the geoparquet details are known.